packages/gamut/src/ToolTip/index.tsx

```diff
@@ -1,15 +1,162 @@
+
+const TooltipWrapper = styled.div`
+  position: relative;
+  display: inline-flex;
+`;
+
+const TargetContainer = styled.div`
+  cursor: pointer;
+  border: 0;
+  background: none;
+  padding: 0;
+`;
+
+type ToolTipContainerProps = {
+  position: ToolTipPosition;
+  variant: VisualTheme;
+};
+
+const ToolTipContainer = styled.div<ToolTipContainerProps>`
+  ${fontSmoothing()}
+
+  display: flex;
+  opacity: 0;
+  transition: opacity 0ms;
+  transition-delay: 750ms;
+  position: absolute;
+  width: 16rem;
+  visibility: hidden;
+
+  ${({ theme, variant }) =>
+    variant === 'dark' && `background: ${theme.colors.black};`}
+
+  // Both before and after psuedo-elements are used because ::after's background should go over the container's
+  // and ::before's box-shadow should be behind the container itself
+  &::after,
+  &::before {
+    content: '';
+    display: block;
+    height: ${arrowHeight};
+    position: absolute;
+    transform: rotate(45deg);
+    width: ${arrowHeight};
+  }
+
+  &::after {
+    background: ${({ theme, variant }) =>
+      variant === 'dark' ? theme.colors.black : theme.colors.white};
+  }
+
+  ${TargetContainer}:hover + &,
+  ${TargetContainer}:focus + &,
+  &:hover {
+    opacity: 1;
+    visibility: visible;
+  }
+
+  ${({ position }) =>
+    ['top-left', 'top-right'].includes(position) &&
+    `
+      bottom: ${containerOffsetVertical};
+
+      &::after,
+      &::before {
+        bottom: ${arrowOffsetVertical};
+      }
+
+      &::before {
+        box-shadow: 2px 2px 4px 0 ${slightShadow};
+      }
+
+      ${ToolTipBody} {
+        box-shadow: 0 2px 4px 0 ${slightShadow};
+      }
+    `}
+
+  ${({ position }) =>
+    ['bottom-left', 'bottom-right'].includes(position) &&
+    `
+      top: ${containerOffsetVertical};
+
+      &::after,
+      &::before {
+        top: ${arrowOffsetVertical};
+      }
+
+      &::before {
+        box-shadow: -2px -2px 4px 0 ${slightShadow};
+      }
+
+      ${ToolTipBody} {
+        box-shadow: 0 0 4px 0 ${slightShadow};
+      }
+    `}
+
+${({ position }) =>
+    ['bottom-left', 'top-left'].includes(position) &&
+    `
+      justify-content: flex-end;
+      right: $container-offset-horizontal;
+
+      &::after,
+      &::before {
+        right: ${arrowOffsetHorizontal};
+      }
+    `}
+
+${({ position }) =>
+    ['bottom-right', 'top-right'].includes(position) &&
+    `
+  left: ${containerOffsetHorizontal};
+
+      &::after,
+      &::before {
+        left: ${arrowOffsetHorizontal};
+      }
+    `}
+`;
+
+const ToolTipBody = styled.div<{ variant: VisualTheme }>`
+  display: inline-block;
+  font-size: ${pxRem(14)};
+  line-height: ${pxRem(18)};
+  padding: 0.6rem 0.75rem;
+
+  ${({ theme, variant }) =>
+    variant === 'light'
+      ? `
+      background: ${theme.colors.white};
+      color: ${theme.colors.black};
+      `
+      : `
+      background: ${theme.colors.black};
+      color: ${theme.colors.white};
+      `}
```

Comment:
Since we're not using it outside of `ToolTipContainer` maybe: 

```suggestion
      background: inherit;
      color: inherit;
```